### PR TITLE
Update Storage to track 2 SDK

### PR DESCRIPTION
**Pull request by FrankieTF**
## Purpose:
- Update Storage to track 2 SDK.

## Updated projects (others not related to T1 SDK):
- 3-WebApp-multi-APIs
  - Updated Storage, Microsoft.WindowsAzure.Storage to  Azure.Storage.Blobs 
- 4-WebApp-your-API\4-2-B2C
  - Only removed Microsoft.WindowsAzure.Storage package since it's never used

[There is a similar repo updated Storage to track 2 SDK](https://github.com/Azure-Samples/storage-dotnet-azure-ad-msal)

### CODE REVIEW

File: 3-WebApp-multi-APIs/Controllers/HomeController.cs [hunk 1]
```diff
@@ -74,21 +74,26 @@ public async Task<IActionResult> Tenants()
...
+            byte[] byteArray = Encoding.ASCII.GetBytes(blobContents);
+            using (MemoryStream stream = new MemoryStream(byteArray))
+            {
+                try
+                {
+                    await blobClient.UploadAsync(stream);
+                    message = "Blob successfully created";
+                }
+                catch (Exception)
+                {
```
Comment (TiagoBrenck) [inline 3-WebApp-multi-APIs/Controllers/HomeController.cs:92]: If you sign-in, restart the application, and go to Blog page again, `MsalUiRequiredException` is thrown and the attribute on top of the ActionResult tries to handle it. For that, you cannot swallow the exception, so I recommend adding this:

```c#
catch(MsalUIRequiredException ex)
{
   throw ex;
}
```

**Review by TiagoBrenck — CHANGES_REQUESTED** [review]
Even after changing the `try catch` I was not able to get this sample to work. The error that I am getting is:

```
OpenIdConnectProtocolException: Message contains error: 'invalid_client', error_description: 'AADSTS650052: The app needs access to a service (\"https://*.blob.core.windows.net\") that your organization \"da0cc090-dd32-4d9f-984c-130b337f221c\" has not subscribed to or enabled. Contact your IT Admin to review the configuration of your service subscriptions. Trace ID: 3fff6b63-7c9a-4d13-bbdd-bd648e7f0900 Correlation ID: fac325fa-7b75-4ca6-84d6-1e69a8e387ab Timestamp: 2020-09-25 19:47:40Z', error_uri: 'https://login.microsoftonline.com/error?code=650052'.
```

@jmprieur I dont know if this issue is on M.I.W side, on the `AuthorizeForScopes`

## Steps taken

1. Register the app using AppCreation scripts
1. Sign-in with a user
1. Stop the application and start it again
1. Navigate to Blob page

File: 3-WebApp-multi-APIs/Controllers/HomeController.cs (hunk 1 above)
Comment (FrankieTF) [reply 3-WebApp-multi-APIs/Controllers/HomeController.cs:97]: Updated.

**Comment by jongio** [comment]
@TiagoBrenck - Can you please try again?
@jmprieur - Is there anyone else you'd like to review this?

**Comment by TiagoBrenck** [comment]
@jongio even after changing the try catch I was not able to get this sample to work. The `try catch` is definitely a must, but I got the error: 

```
OpenIdConnectProtocolException: Message contains error: 'invalid_client', error_description: 'AADSTS650052: The app needs access to a service (\"https://*.blob.core.windows.net\") that your organization \"da0cc090-dd32-4d9f-984c-130b337f221c\" has not subscribed to or enabled. Contact your IT Admin to review the configuration of your service subscriptions. Trace ID: 3fff6b63-7c9a-4d13-bbdd-bd648e7f0900 Correlation ID: fac325fa-7b75-4ca6-84d6-1e69a8e387ab Timestamp: 2020-09-25 19:47:40Z', error_uri: 'https://login.microsoftonline.com/error?code=650052'.
```

If you follow my steps mentioned [here](https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/pull/413#pullrequestreview-496768491), could you make it to work?

**Comment by jongio** [comment]
@TiagoBrenck - Can you please try to run this sample without this PR?  We want to make sure it is runable, because the updates we are doing are just to upgrade the libraries, which shouldn't cause the failure you are seeing above.  So, basically, try to run from master please and let us know if you can get it to run.

**Review by Shama-K — APPROVED** [review]
Tested and it worked fine.
Thanks.

File: 3-WebApp-multi-APIs/Startup.cs [hunk 2]
```diff
@@ -40,11 +38,10 @@ public void ConfigureServices(IServiceCollection services)
...
             });
 
-            services.AddOptions();
-
-            services.AddMicrosoftIdentityWebAppAuthentication(Configuration)
-                    .EnableTokenAcquisitionToCallDownstreamApi( new string[] { Constants.ScopeUserRead })
-                    .AddInMemoryTokenCaches();
+            services.AddAuthentication(OpenIdConnectDefaults.AuthenticationScheme)
+                .AddMicrosoftIdentityWebApp(Configuration.GetSection("AzureAd"))
+                .EnableTokenAcquisitionToCallDownstreamApi(null)
```
Comment (jmprieur) [inline 3-WebApp-multi-APIs/Startup.cs:43]: the null should not be necessary. Is it?

File: 3-WebApp-multi-APIs/WebApp-OpenIDConnect-DotNet.csproj [hunk 3]
```diff
@@ -18,9 +18,10 @@
   </ItemGroup>
 
   <ItemGroup>
+    <PackageReference Include="Azure.Storage.Blobs" Version="12.6.0" />
+    <PackageReference Include="Microsoft.Identity.Client" Version="4.19.0" />
     <PackageReference Include="Microsoft.Identity.Web" Version="0.4.0-preview" />
```
Comment (jmprieur) [inline 3-WebApp-multi-APIs/WebApp-OpenIDConnect-DotNet.csproj:23]: you'll have a merge conflict here. Microsoft.Identity.Web is 1.0.0 now

**Review by jmprieur — APPROVED** [review]
LGTM.
Thanks @FrankieTF and @Shama-K

File: 3-WebApp-multi-APIs/Startup.cs (hunk 2 above)
Comment (FrankieTF) [reply 3-WebApp-multi-APIs/Startup.cs]: Yes, removed.

File: 3-WebApp-multi-APIs/WebApp-OpenIDConnect-DotNet.csproj (hunk 3 above)
Comment (FrankieTF) [reply 3-WebApp-multi-APIs/WebApp-OpenIDConnect-DotNet.csproj]: Updated to latest GA package.

**Pull request merged**